pybamm/solvers/processed_variable.py

```diff
@@ -297,40 +275,43 @@ def initialise_2D(self):
         self.second_dim_pts = second_dim_pts
 
         # set up interpolation
-        self._interpolation_function = interp.RegularGridInterpolator(
-            (first_dim_pts, second_dim_pts, self.t_sol),
-            entries,
-            method="linear",
-            fill_value=np.nan,
-        )
-
-    def initialise_2D_fixed_t_scikit_fem(self):
-        y_sol = self.mesh[0].edges["y"]
-        len_y = len(y_sol)
-        z_sol = self.mesh[0].edges["z"]
-        len_z = len(z_sol)
-
-        # Evaluate the base_variable
-        inputs = {name: inp[0] for name, inp in self.inputs.items()}
-
-        entries = np.reshape(
-            self.base_variable.evaluate(0, self.u_sol, inputs=inputs), [len_y, len_z]
-        )
+        if len(self.t_sol) == 1:
+            # function of space only. Note the order of the points is the reverse
+            # of what you'd expect
+            interpolant = interp.interp2d(
+                second_dim_pts,
+                first_dim_pts,
+                entries[:, :, 0],
+                kind="linear",
+                fill_value=np.nan,
+            )
 
-        # assign attributes for reference
-        self.entries = entries
-        self.dimensions = 2
-        self.y_sol = y_sol
-        self.z_sol = z_sol
-        self.first_dimension = "y"
-        self.second_dimension = "z"
-        self.first_dim_pts = y_sol
-        self.second_dim_pts = z_sol
+            def interp_fun(input):
+                first_dim, second_dim, _ = input
+                if isinstance(first_dim, np.ndarray) and isinstance(
+                    second_dim, np.ndarray
+                ):
+                    first_dim = first_dim[:, 0, 0]
+                    second_dim = second_dim[:, 0]
+                    return interpolant(second_dim, first_dim)
+                elif isinstance(first_dim, np.ndarray):
+                    first_dim = first_dim[:, 0]
+                    return interpolant(second_dim, first_dim)[:, 0]
+                elif isinstance(second_dim, np.ndarray):
+                    second_dim = second_dim[:, 0]
+                    return interpolant(second_dim, first_dim)
+                else:
+                    return interpolant(second_dim, first_dim)[0]
```

Comment:
is there no way around the duplication of all these lines?

Reply:
yep, have made a function to take the interpolant and inputs and returns the result of the correct shape